### PR TITLE
Minor fixes

### LINKED ISSUE #2213: Builder creates too much particles
### Minecolonies version

minecolonies-universal-1.12.2-0.8.7508

### Expected behavior

- Creates some particles like if you would be in process of breaking block

### Actual behaviour

- Creates particles on every hit like if you would already break the block.

### Steps to reproduce the problem

1. Get builder to build anything that requires to break block
2. On every hit he/she will create too much particles

3. This could affect on perfomance of weak PCs
4. Profit

### LINKED ISSUE #2217: [1.12.2] Lumberjack cuts down structures with logs in them.
### Version Info

Minecolonies: 1.12.2 - 08.7508
Forge:  14.23.2.2611

### Expected behavior

- Lumberjack ignores structures with huts/ structures in general.

### Actual behaviour

- Lumberjack chops log blocks of town structures near the sawmill.

### Steps to reproduce the problem

Unsure really, I just was wondering why one of the town structures was repaired yet broken, and also whilst being qued for repair the lumberjack jack in the town ran up to the structure and started cutting down all the logs and planting saplings as it usually would for trees, the structure was probably around 30 or fewer blocks away from the sawmill and it was the closest thing near the sawmill with logs in it.

May have been a glitch it has not done this again yet since the structure was repaired and I planted some extra trees for the little guy to go too instead of the structures.

### LINKED ISSUE #2219: Suggestion: Report location of the block that can't be broken with tool in inventory
I have seen several people complain about the miner requesting a diamond pick axe before it is time for them to be able to use one. This includes me. It would be helpful if in the details of the min/max of pick axe level needed it could also report the coords of the block it can not break with a current tool that is in their inventory.

### LINKED ISSUE #2223: [1.12.2] NPE While loading existing world. (minecolonies.structures.helpers.Structure.getSize)
###  Version Info
Minecolonies  ` 1.12.2-0.8.7508`
Forge `14.23.2.2611`  

### Expected behaviour
- Normal World Load and continue playing

### Actual behaviour
- World Loading is looping on [Loading World](https://i.imgur.com/yi3qrII.png) screen and never initiates `Building Terrain`
upon closing out of the game and trying to load the same world again it will load fine but the colony data is completely erased and no colonies exist in the world (it's like this mod self-deleted the colony due to the error). 

[Here is the crash-log!](https://pastebin.com/LF5vDq1m)

After force closing the java process and loading MC back up again I can enter the world w/o problem except the colony is gone -

Thankfully this mod makes automated backups colony data however whenever I replace the data with that of a backup made two days ago the problem above occurs again. If no colony exists the world loads normally like below. 

### Extended Information
This is a world I have been playing on for the past few weeks and properly closed out of a few days ago to have a break from MC and went to continue playing again today! 

Oddly enough the only mods which I updated were Pams Harvest Craft & Reborn Core & Biomes O Plenty, though I don't think this would have had any effect as the update was to a core mod for StevesCarts Reborn (which I have no entity or tile data for that mod in this world yet, and Pam's Harvest added some new recipes requiring a config regen for HarvestCraft.cfg. Biomes O Plenty updated a simple command fix to correctly tpbiome, I am aware that if they changed world-gen or something it may be an issue, however, the world loads correctly without the colony existing,

### Steps to reproduce the problem
I don't know sorry, I didn't do anything in-game to cause this I just loaded up an existing world after exiting the game a few a days ago.

### LINKED ISSUE #2224: Build tool GUI crashes on Forge version 2615
### Minecolonies version

version 0.8.7508

### Expected behavior

- The build tool gui works

### Actual behaviour

- It doesn't, causing a crash when using both the build tool, supply ship and supply camp tools.

### Steps to reproduce the problem

1. Download forge latest, version 2615
2. Realize the latest forge version has a changelog of "Clean up some GUI code and change magic numbers to GL constants"
3. Look at the log here: https://pastebin.com/0FuvW9r7 and the crash report here: https://pastebin.com/qHiATG7T
4. Find that the number changes in the latest forge version causes the array in the build tool GUI to attempt to subtract a number from a nil value

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #2223
Closes #2224
Closes #2219
Closes #2219
Closes #2213
Closes #2217

# Changes proposed in this pull request:
- Fixes constructiontape being placed on workLoad
- Fixes issues with deleted structures
- Protects buildings from being detected as trees
- Reduces particle effects
- Reports block position if tool is not good enough.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 1]
```diff
@@ -1720,16 +1720,12 @@ public void onRequestCancelled(@NotNull final IRequestManager manager, @NotNull
...
         {
-            if (getOpenRequestsByRequestableType().containsKey(TypeToken.of(requestThatCompleted.getRequest().getClass())))
+            getOpenRequestsByRequestableType().get(TypeToken.of(requestThatCompleted.getRequest().getClass())).remove(token);
+            if (getOpenRequestsByRequestableType().get(TypeToken.of(requestThatCompleted.getRequest().getClass())).isEmpty())
             {
-                getOpenRequestsByRequestableType().get(TypeToken.of(requestThatCompleted.getRequest().getClass())).remove(token);
-
-                if (getOpenRequestsByRequestableType().get(TypeToken.of(requestThatCompleted.getRequest().getClass())).isEmpty())
-                {
                     getOpenRequestsByRequestableType().remove(TypeToken.of(requestThatCompleted.getRequest().getClass()));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MINOR Make this line start at column 17. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 89 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 71 major
* ![MINOR][MINOR] 9 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L230](https://github.com/Minecolonies/minecolonies/blob/4fc4022537ba1ccee6480cb3091a2702238fe7f2/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L230): Remove this call from a constructor to the overridable "setupRsDataStore" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1367](https://github.com/Minecolonies/minecolonies/blob/4fc4022537ba1ccee6480cb3091a2702238fe7f2/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1367): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1437](https://github.com/Minecolonies/minecolonies/blob/4fc4022537ba1ccee6480cb3091a2702238fe7f2/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1437): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1468](https://github.com/Minecolonies/minecolonies/blob/4fc4022537ba1ccee6480cb3091a2702238fe7f2/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1468): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1480](https://github.com/Minecolonies/minecolonies/blob/4fc4022537ba1ccee6480cb3091a2702238fe7f2/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1480): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1583](https://github.com/Minecolonies/minecolonies/blob/4fc4022537ba1ccee6480cb3091a2702238fe7f2/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1583): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/4fc4022537ba1ccee6480cb3091a2702238fe7f2/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/4fc4022537ba1ccee6480cb3091a2702238fe7f2/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/4fc4022537ba1ccee6480cb3091a2702238fe7f2/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/4fc4022537ba1ccee6480cb3091a2702238fe7f2/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java [hunk 2]
```diff
@@ -1720,16 +1720,12 @@ public void onRequestCancelled(@NotNull final IRequestManager manager, @NotNull
...
+            getOpenRequestsByRequestableType().get(TypeToken.of(requestThatCompleted.getRequest().getClass())).remove(token);
+            if (getOpenRequestsByRequestableType().get(TypeToken.of(requestThatCompleted.getRequest().getClass())).isEmpty())
             {
-                getOpenRequestsByRequestableType().get(TypeToken.of(requestThatCompleted.getRequest().getClass())).remove(token);
-
-                if (getOpenRequestsByRequestableType().get(TypeToken.of(requestThatCompleted.getRequest().getClass())).isEmpty())
-                {
-                    getOpenRequestsByRequestableType().remove(TypeToken.of(requestThatCompleted.getRequest().getClass()));
-                }
+                getOpenRequestsByRequestableType().remove(TypeToken.of(requestThatCompleted.getRequest().getClass()));
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java]: MINOR Make this line start at column 17. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AIndentationCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java [hunk 3]
```diff
@@ -3,6 +3,7 @@
 import com.minecolonies.api.configuration.Configurations;
 import com.minecolonies.api.util.*;
 import com.minecolonies.coremod.blocks.ModBlocks;
+import com.minecolonies.coremod.colony.WorkManager;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java]: MINOR Remove this unused import 'com.minecolonies.coremod.colony.WorkManager'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java [hunk 4]
```diff
@@ -606,11 +607,30 @@ public void loadStructure(@NotNull final String name, final int rotateTimes, fin
...
             }
 
-            ((AbstractJobStructure) job).getStructure().rotate(rotateTimes, world, position, isMirrored ? Mirror.FRONT_BACK : Mirror.NONE);
-            ((AbstractJobStructure) job).getStructure().setPosition(position);
+            try
+            {
+                ((AbstractJobStructure) job).getStructure().rotate(rotateTimes, world, position, isMirrored ? Mirror.FRONT_BACK : Mirror.NONE);
+                ((AbstractJobStructure) job).getStructure().setPosition(position);
+            }
+            catch(final NullPointerException ex)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java:614]: MAJOR Avoid catching NullPointerException. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1696)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 91 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 72 major
* ![MINOR][MINOR] 10 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L230](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L230): Remove this call from a constructor to the overridable "setupRsDataStore" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1367](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1367): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1437](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1437): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1468](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1468): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1480](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1480): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1583](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1583): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 90 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 72 major
* ![MINOR][MINOR] 9 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L230](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L230): Remove this call from a constructor to the overridable "setupRsDataStore" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1367](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1367): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1437](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1437): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1468](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1468): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1480](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1480): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1583](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1583): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/a625678e18fae0aa74f865a9428caaab63b63cb6/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 89 issues
* ![CRITICAL][CRITICAL] 6 critical
* ![MAJOR][MAJOR] 72 major
* ![MINOR][MINOR] 8 minor
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L230](https://github.com/Minecolonies/minecolonies/blob/f57354f56ad74bd23e335986af1594faf16ab653/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L230): Remove this call from a constructor to the overridable "setupRsDataStore" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1699)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1367](https://github.com/Minecolonies/minecolonies/blob/f57354f56ad74bd23e335986af1594faf16ab653/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1367): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1437](https://github.com/Minecolonies/minecolonies/blob/f57354f56ad74bd23e335986af1594faf16ab653/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1437): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1468](https://github.com/Minecolonies/minecolonies/blob/f57354f56ad74bd23e335986af1594faf16ab653/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1468): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1480](https://github.com/Minecolonies/minecolonies/blob/f57354f56ad74bd23e335986af1594faf16ab653/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1480): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![CRITICAL][CRITICAL] [AbstractBuilding.java#L1583](https://github.com/Minecolonies/minecolonies/blob/f57354f56ad74bd23e335986af1594faf16ab653/src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java#L1583): Remove usage of generic wildcard type. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1452)
1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/f57354f56ad74bd23e335986af1594faf16ab653/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/f57354f56ad74bd23e335986af1594faf16ab653/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/f57354f56ad74bd23e335986af1594faf16ab653/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/f57354f56ad74bd23e335986af1594faf16ab653/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**